demos/demo_solo12_from_yaml.py

```diff
@@ -1,55 +1,48 @@
 #! /usr/bin/env python
 
+import libodri_control_interface_pywrap as oci
 import numpy as np
 
 np.set_printoptions(suppress=True, precision=2)
 
-import time
-
-import libmaster_board_sdk_pywrap as mbs
-import libodri_control_interface_pywrap as oci
-
-import pathlib
-
 robot = oci.robot_from_yaml_file("config_solo12.yaml")
 joint_calibrator = oci.joint_calibrator_from_yaml_file(
     "config_solo12.yaml", robot.joints
 )
 
 # Initialize the communication and the session.
 robot.start()
-robot.wait_until_ready()
+if not robot.wait_until_ready():
+    raise RuntimeError(
+        "Error during the waiting time, please check the error messages.")
 
-# Calibrate the robot if needed.
-robot.run_calibration(joint_calibrator)
+# Calibrate the robot if needed and get into initial configuration
+initial_configuration = np.array(
+    [0.0, 0.7, -1.4, -0.0, 0.7, -1.4, 0.0, -0.7, +1.4, -0.0, -0.7, +1.4])
+if not robot.run_calibration(joint_calibrator, initial_configuration):
+    raise RuntimeError(
+        "Error during the calibration, please check the error messages.")
 
-robot.parse_sensor_data()
-
-
-init_imu_attitude = robot.imu.attitude_euler.copy()
-
-des_pos = np.zeros(12)
+robot.joints.set_zero_commands()
 
 c = 0
-dt = 0.001
-calibration_done = False
-next_tick = time.time()
 while not robot.is_timeout:
     robot.parse_sensor_data()
 
-    imu_attitude = robot.imu.attitude_euler
-    positions = robot.joints.positions
-    velocities = robot.joints.velocities
-
-    if c % 2000 == 0:
-        print("IMU attitude:", imu_attitude)
-        print("joint pos   :", positions)
-        print("joint vel   :", velocities)
+    if c % 100 == 0:
+        print("IMU attitude:", robot.imu.attitude_euler)
+        print("joint pos   :", robot.joints.positions)
+        print("joint vel   :", robot.joints.velocities)
         robot.robot_interface.PrintStats()
 
-    des_pos[:] = imu_attitude[2] - init_imu_attitude[2]
-    torques = 5.0 * (des_pos - positions) - 0.1 * velocities
-    robot.joints.set_torques(torques)
+    # Waiting in initial configuration
+    robot.joints.set_torques(np.zeros(12))
+    robot.joints.set_desired_positions(initial_configuration)
+    robot.joints.set_desired_velocities(np.zeros(12))
+    robot.joints.set_position_gains(3.0 * np.ones(12))
+    robot.joints.set_velocity_gains(0.05 * np.ones(12))
```

Comment:
Similar like in the C++ example, can you allocate the `np.zeros(12)` outside the loop to avoid allocating them on every control cycle?